webapp/channels/src/components/user_settings/general/user_settings_general.tsx

```diff
@@ -165,14 +164,8 @@ export class UserSettingsGeneralTab extends PureComponent<Props, State> {
     }
 
     componentDidMount() {
-        if (this.props.enableCustomProfileAttributes) {
-            const fetchValues = async () => {
-                const response = await Client4.getUserCustomProfileAttributesValues(this.props.user.id);
-                this.setState({customAttributeValues: response});
-            };
-
-            this.props.actions.getCustomProfileAttributeFields();
-            fetchValues();
+        if (this.props.enableCustomProfileAttributes && !this.props.user.custom_profile_attributes) {
+            this.props.actions.getCustomProfileAttributeValues(this.props.user.id);
```

Comment:
This can be removed when initial profile is retrieved with custom profille attributes. Currently, it will only be called once.